### PR TITLE
susie using summary statistics Xty, XtX

### PR DESCRIPTION
**Pull request by zouyuxin**
I add function `update_each_effect_ss`, `single_effect_regression_ss`, and `susie_ss` to fit the model.
The KL and elbo are deleted from the output.

### CODE REVIEW

File: R/susie_ss.R [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#' n = 1000
+#' p = 1000
+#' beta = rep(0,p)
+#' beta[1] = 1
+#' beta[2] = 1
+#' beta[3] = 1
+#' beta[4] = 1
+#' X = matrix(rnorm(n*p),nrow=n,ncol=p)
+#' y = X %*% beta + rnorm(n)
+#' res =susie_ss(XtX,Xty,L=10)
```
Comment (stephens999) [inline R/susie_ss.R]: i think you need to define these quantities?

File: R/susie_ss.R [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+    # Check inputs sigma and sa.
+    if (length(residual_variance) != 1)
+      stop("Inputs residual_variance must be scalar")
+    # Check inputs sigma and sa.
+    if (length(prior_variance) != L)
+      stop("Inputs prior_variance must be of length 1 or L")
+
+    #initialize susie fit
+    s = list(alpha=matrix(1/p,nrow=L,ncol=p), mu=matrix(0,nrow=L,ncol=p),
```
Comment (stephens999) [inline R/susie_ss.R]: i think best to include s$intercept = 0 so this way the coef(s) will be the same length for both ss and non-ss versions.

**Comment by stephens999** [comment]
i added a couple of comments on code. Do you see them?

Do I gather you also have an implementation that estimates the prior_variance parameter as well? Can you add a test to show that works?

**Comment by zouyuxin** [comment]
The output from susie_ss includes XtXr, which is t(X)%*%fitted_value. The original susie's output has fitted value, to use in predict function. Here we don't have fitted value, do we need to include XtXr as output?

**Comment by zouyuxin** [comment]
You didn't say about the implementation of estimating the prior_variance. I change the function about the estimate_prior_variance to make it works. But I don't know the underlying method.

**Pull request merged**